### PR TITLE
SPUI-1429 > Banners bottom sheet fixes

### PR DESCRIPTION
**Pull request by Anton-Tesenchuk**

### CODE REVIEW

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/earn_bottom_sheet.dart [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+        pinnedSmall: const EarnPinnedSmall(),
+        expandedHeight: 320,
+        children: [
+          EarnBody(
+            onTap: onTap,
+          ),
+        ],
+      );
+    },
+    transitionAnimationController: null,
```
Comment (VovaMedvediev) [inline lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/earn_bottom_sheet.dart:33]: I think this parameter shouldn't be required if it's nullable

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_pinned.dart [hunk 2]
```diff
@@ -1,35 +1,72 @@
...
+              right: 24.0,
+              child: GestureDetector(
+                onTap: () => Navigator.pop(context),
+                child: const SEraseMarketIcon(),
               ),
-              fit: BoxFit.fill,
             ),
-          ),
+            Positioned(
+              width: MediaQuery.of(context).size.width,
```
Comment (VovaMedvediev) [inline lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_pinned.dart:45]: use mediaQueryPod instead please

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+      Navigator.pop(context);
+    }
+
+    return WillPopScope(
+        onWillPop: () {
+          _onDissmisAction(context);
+          return Future.value(true);
+        },
+        child: Padding(
+          padding: MediaQuery.of(context).viewInsets,
```
Comment (VovaMedvediev) [inline lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart:113]: mediaQueryPod

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+  @override
+  _EarnBottomSheetContainerState createState() =>
+      _EarnBottomSheetContainerState();
+}
+
+class _EarnBottomSheetContainerState extends State<EarnBottomSheetContainer> {
+  late double _offset;
+  late bool isAnimatingNow;
+  late bool isClosing;
+  final controller = ScrollController();
```
Comment (VovaMedvediev) [inline lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart:49]: use useScrollController instead - https://stackoverflow.com/questions/63832051/safe-usage-for-usescrollcontroller-flutter-hooks.
Actually, you don't need StatefulWidget here.

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+  }
+
+  void _setOffset() {
+    if (controller.hasClients) {
+      if (_offset > controller.offset &&
+          !isAnimatingNow &&
+          controller.offset < controller.position.maxScrollExtent
+      ) {
+        setState(() {
+          isAnimatingNow = true;
```
Comment (VovaMedvediev) [inline lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart:72]: final isAnimatingNow = useState(true) can be used instead

**Review by VovaMedvediev — CHANGES_REQUESTED** [review]

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart (hunk 3 above)
Comment (Anton-Tesenchuk) [reply lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart]: Done

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+                                      children: [
+                                        ...widget.children,
+                                      ],
+                                    ),
+                                  ),
+                                ],
+                              ),
+                              Positioned(
+                                child: Container(
+                                  width: MediaQuery.of(context).size.width - 44,
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_bottom_sheet_container.dart]: Done

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_pinned.dart [hunk 7]
```diff
@@ -1,35 +1,72 @@
...
             ),
-            image: DecorationImage(
-              image: AssetImage(
-                earnGroupImageAsset,
+            Positioned(
+              top: 24.0,
+              right: 24.0,
+              child: GestureDetector(
+                onTap: () => Navigator.pop(context),
+                child: const SEraseMarketIcon(),
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/components/earn_pinned.dart]: Done

File: lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/earn_bottom_sheet.dart (hunk 1 above)
Comment (Anton-Tesenchuk) [reply lib/app/screens/portfolio/view/components/empty_portfolio/components/earn_bottom_sheet/earn_bottom_sheet.dart]: Done

**Review by VovaMedvediev — APPROVED** [review]

**Pull request merged**